static_frame/core/index_hierarchy.py

```diff
@@ -1603,184 +2305,209 @@ def loc_searchsorted(self,
+            self._update_array_cache() # pragma: no cover
 
         return constructor(
                 self._blocks.copy(),
                 columns=None,
                 index=None,
-                own_data=True
+                own_data=True,
                 )
 
-    def to_frame(self) -> 'Frame':
+    def to_frame(self: IH) -> 'Frame':
         '''
-        Return :obj:`Frame` version of this :obj:`IndexHiearchy`.
+        Return :obj:`Frame` version of this :obj:`IndexHierarchy`.
         '''
         from static_frame import Frame
         return self._to_frame(Frame)
 
-    def to_frame_go(self) -> 'FrameGO':
+    def to_frame_go(self: IH) -> 'FrameGO':
         '''
         Return a :obj:`FrameGO` version of this :obj:`IndexHierarchy`.
         '''
         from static_frame import FrameGO
-        return self._to_frame(FrameGO) #type: ignore
+        return tp.cast(FrameGO, self._to_frame(FrameGO))
 
-    def to_pandas(self) -> 'DataFrame':
-        '''Return a Pandas MultiIndex.
+    def to_pandas(self: IH) -> 'DataFrame':
+        '''
+        Return a Pandas MultiIndex.
         '''
         import pandas
 
         if self._recache:
-            self._update_array_cache()
+            self._update_array_cache() # pragma: no cover
 
         # must copy to get a mutable array
-        arrays = tuple(a.copy() for a in self._blocks.axis_values(axis=0))
-        mi = pandas.MultiIndex.from_arrays(arrays)
-
+        mi = pandas.MultiIndex(
+                levels=[index.values.copy() for index in self._indices],
+                codes=[arr.copy() for arr in self._indexers],
+                )
         mi.name = self._name
         mi.names = self.names
         return mi
 
-    def to_tree(self) -> TreeNodeT:
-        '''Returns the tree representation of an IndexHierarchy
+    def _build_tree_at_depth_from_mask(self: IH,
+            depth: int,
+            mask: np.ndarray,
+            ) -> tp.Union[TreeNodeT, Index]:
+        '''
+        Recursively build a tree of :obj:`TreeNodeT` at `depth` given `mask`
+        '''
+        # This private internal method assumes recache has already been checked for!
+
+        index_at_depth = self._indices[depth]
+        indexer_at_depth = self._indexers[depth]
+
+        if depth == self.depth - 1:
+            values = index_at_depth[indexer_at_depth[mask]]
+            return index_at_depth.__class__(values)
+
+        tree: TreeNodeT = {}
+
+        for i in ufunc_unique(indexer_at_depth[mask]):
+            tree[index_at_depth[i]] = self._build_tree_at_depth_from_mask(
+                    depth=depth + 1,
+                    mask=mask & (indexer_at_depth == i),
+                    )
+
+        return tree
+
+    def to_tree(self: IH) -> TreeNodeT:
         '''
-        return self._levels.traverse() # type: ignore
+        Returns the tree representation of an IndexHierarchy
+        '''
+        if self._recache:
+            self._update_array_cache() # pragma: no cover
```

Comment:
Why no coverage?

Reply:
See this response: https://github.com/InvestmentSystems/static-frame/pull/437#discussion_r831501537